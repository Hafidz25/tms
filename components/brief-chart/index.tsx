import BriefChart from "./brief-chart";

export { BriefChart };
