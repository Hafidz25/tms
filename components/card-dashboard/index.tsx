import CardDashboard from "./card-dashboard";

export { CardDashboard };
