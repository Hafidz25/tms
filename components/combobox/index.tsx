export * from './combobox';
export { type ComboboxOption } from './use-combobox';