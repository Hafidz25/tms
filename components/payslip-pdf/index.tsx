import PayslipPdf from "./payslip-pdf";

export { PayslipPdf };
