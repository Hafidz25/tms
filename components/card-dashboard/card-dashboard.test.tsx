import { describe, expect, it, test } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import { CardDashboard } from "@/components/card-dashboard";

describe("Card Dashboard Component", () => {
  it("render card dashboard component", () => {
    const component = render(
      //@ts-ignore
      <CardDashboard userExist={null} users={[]} briefs={[]} />
    );
    expect(component).toMatchSnapshot();
  });
});
