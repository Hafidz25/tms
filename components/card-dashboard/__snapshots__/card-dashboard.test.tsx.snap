// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card Dashboard Component > render card dashboard component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="grid gap-4 md:grid-cols-2 md:gap-8 lg:grid-cols-4"
      >
        <div
          class="rounded-lg border border-slate-200 bg-white text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50"
          x-chunk="dashboard-01-chunk-0"
        >
          <div
            class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
          >
            <h3
              class="tracking-tight text-sm font-medium"
            >
              Customer Services
            </h3>
            <svg
              class="lucide lucide-headset h-4 w-4 text-muted-foreground"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 11h3a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-5Zm0 0a9 9 0 1 1 18 0m0 0v5a2 2 0 0 1-2 2h-1a2 2 0 0 1-2-2v-3a2 2 0 0 1 2-2h3Z"
              />
              <path
                d="M21 16v2a4 4 0 0 1-4 4h-5"
              />
            </svg>
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="text-2xl font-bold"
            >
              0
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border border-slate-200 bg-white text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50"
          x-chunk="dashboard-01-chunk-1"
        >
          <div
            class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
          >
            <h3
              class="tracking-tight text-sm font-medium"
            >
              Team Member
            </h3>
            <svg
              class="lucide lucide-users h-4 w-4 text-muted-foreground"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
              />
              <circle
                cx="9"
                cy="7"
                r="4"
              />
              <path
                d="M22 21v-2a4 4 0 0 0-3-3.87"
              />
              <path
                d="M16 3.13a4 4 0 0 1 0 7.75"
              />
            </svg>
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="text-2xl font-bold"
            >
              0
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border border-slate-200 bg-white text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50"
          x-chunk="dashboard-01-chunk-2"
        >
          <div
            class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
          >
            <h3
              class="tracking-tight text-sm font-medium"
            >
              Brief Ongoing
            </h3>
            <svg
              class="lucide lucide-briefcase-business h-4 w-4 text-muted-foreground"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12 12h.01"
              />
              <path
                d="M16 6V4a2 2 0 0 0-2-2h-4a2 2 0 0 0-2 2v2"
              />
              <path
                d="M22 13a18.15 18.15 0 0 1-20 0"
              />
              <rect
                height="14"
                rx="2"
                width="20"
                x="2"
                y="6"
              />
            </svg>
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="text-2xl font-bold"
            >
              0
            </div>
          </div>
        </div>
        <div
          class="rounded-lg border border-slate-200 bg-white text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50"
          x-chunk="dashboard-01-chunk-3"
        >
          <div
            class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
          >
            <h3
              class="tracking-tight text-sm font-medium"
            >
              Brieft Done
            </h3>
            <svg
              class="lucide lucide-package-check h-4 w-4 text-muted-foreground"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m16 16 2 2 4-4"
              />
              <path
                d="M21 10V8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73l7 4a2 2 0 0 0 2 0l2-1.14"
              />
              <path
                d="m7.5 4.27 9 5.15"
              />
              <polyline
                points="3.29 7 12 12 20.71 7"
              />
              <line
                x1="12"
                x2="12"
                y1="22"
                y2="12"
              />
            </svg>
          </div>
          <div
            class="p-6 pt-0"
          >
            <div
              class="text-2xl font-bold"
            >
              0
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="grid gap-4 md:grid-cols-2 md:gap-8 lg:grid-cols-4"
    >
      <div
        class="rounded-lg border border-slate-200 bg-white text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50"
        x-chunk="dashboard-01-chunk-0"
      >
        <div
          class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
        >
          <h3
            class="tracking-tight text-sm font-medium"
          >
            Customer Services
          </h3>
          <svg
            class="lucide lucide-headset h-4 w-4 text-muted-foreground"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 11h3a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-5Zm0 0a9 9 0 1 1 18 0m0 0v5a2 2 0 0 1-2 2h-1a2 2 0 0 1-2-2v-3a2 2 0 0 1 2-2h3Z"
            />
            <path
              d="M21 16v2a4 4 0 0 1-4 4h-5"
            />
          </svg>
        </div>
        <div
          class="p-6 pt-0"
        >
          <div
            class="text-2xl font-bold"
          >
            0
          </div>
        </div>
      </div>
      <div
        class="rounded-lg border border-slate-200 bg-white text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50"
        x-chunk="dashboard-01-chunk-1"
      >
        <div
          class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
        >
          <h3
            class="tracking-tight text-sm font-medium"
          >
            Team Member
          </h3>
          <svg
            class="lucide lucide-users h-4 w-4 text-muted-foreground"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
            />
            <circle
              cx="9"
              cy="7"
              r="4"
            />
            <path
              d="M22 21v-2a4 4 0 0 0-3-3.87"
            />
            <path
              d="M16 3.13a4 4 0 0 1 0 7.75"
            />
          </svg>
        </div>
        <div
          class="p-6 pt-0"
        >
          <div
            class="text-2xl font-bold"
          >
            0
          </div>
        </div>
      </div>
      <div
        class="rounded-lg border border-slate-200 bg-white text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50"
        x-chunk="dashboard-01-chunk-2"
      >
        <div
          class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
        >
          <h3
            class="tracking-tight text-sm font-medium"
          >
            Brief Ongoing
          </h3>
          <svg
            class="lucide lucide-briefcase-business h-4 w-4 text-muted-foreground"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12 12h.01"
            />
            <path
              d="M16 6V4a2 2 0 0 0-2-2h-4a2 2 0 0 0-2 2v2"
            />
            <path
              d="M22 13a18.15 18.15 0 0 1-20 0"
            />
            <rect
              height="14"
              rx="2"
              width="20"
              x="2"
              y="6"
            />
          </svg>
        </div>
        <div
          class="p-6 pt-0"
        >
          <div
            class="text-2xl font-bold"
          >
            0
          </div>
        </div>
      </div>
      <div
        class="rounded-lg border border-slate-200 bg-white text-slate-950 shadow-sm dark:border-slate-800 dark:bg-slate-950 dark:text-slate-50"
        x-chunk="dashboard-01-chunk-3"
      >
        <div
          class="p-6 flex flex-row items-center justify-between space-y-0 pb-2"
        >
          <h3
            class="tracking-tight text-sm font-medium"
          >
            Brieft Done
          </h3>
          <svg
            class="lucide lucide-package-check h-4 w-4 text-muted-foreground"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m16 16 2 2 4-4"
            />
            <path
              d="M21 10V8a2 2 0 0 0-1-1.73l-7-4a2 2 0 0 0-2 0l-7 4A2 2 0 0 0 3 8v8a2 2 0 0 0 1 1.73l7 4a2 2 0 0 0 2 0l2-1.14"
            />
            <path
              d="m7.5 4.27 9 5.15"
            />
            <polyline
              points="3.29 7 12 12 20.71 7"
            />
            <line
              x1="12"
              x2="12"
              y1="22"
              y2="12"
            />
          </svg>
        </div>
        <div
          class="p-6 pt-0"
        >
          <div
            class="text-2xl font-bold"
          >
            0
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
